### ISSUE TITLE
Ingress Status Updates

### ISSUE THREAD

**Issue by jrnt30** [heptio/contour#21]
It was outlined in the Roadmap that Ingress Status updates were coming, just adding an issue for tracking. 

In addition to the example given, there are several projects that rely on this information being present.  Of particular interest to us would be External-DNS, which [utilizes the status](https://github.com/kubernetes-incubator/external-dns/blob/master/source/ingress.go#L183) to determine the appropriate destination for the relevant DNS entries. 

Blocked:

- [ ] #1385

**Comment by davecheney** [comment]
Thanks for raising this issue. I'm going to mark this tentatively for the 0.2 milestone. It doesn't have an ETA as yet, but it'll be the _next_ release.

I know that other ingress controllers go through some effort to ensure that only a single process writes that status information. I haven't looked into the specifics, but I'm hoping it's possible to avoid this complexity with the PATCH method on the API (I know that's a lot of hand waving).

I'm assigning this to myself to come up with a design. It's also going to be blocked by the cache refactor work (PR coming in the next 48 hours) and some discussion of a state machine for converting k8s API events into the envoy configs. Again, this needs a design document.

**Comment by davecheney** [comment]
Bumping to 0.3.0. This is likely to require gRPC to become GA to support it properly.

**Comment by davecheney** [comment]
Bumping to 0.4, sorry

**Comment by davecheney** [comment]
Sorry, once again this is going to get bumped. 0.5, I hope, will resolve it.

**Comment by elisiano** [comment]
@davecheney just out of curiosity: is there an ETA for this? 
0.4.0 is not out yet but I'm noticing this has been bumped to 0.5.0.

**Comment by davecheney** [comment]
I know i've kicked it down the road a few times. I know this is important,
but it's also non trivial, but my desire is to commit to fixing it before
the 0.5 release in may.

**Comment by davecheney** [comment]
With the 0.5 date moved up a month (and a vacation on the team next week) there won't be time to do this for 0.5. Moving to 0.6

**Comment by jrnt30** [comment]
Understood. Thanks for staying on top of the updates, appreciate it!

**Comment by davecheney** [comment]
Sorry. Once again this keeps getting bumped. I've updated the priority to indicate that this is a low priority issue, based on the evidence from the previous releases.

**Comment by jrnt30** [comment]
Do you see any significant changes to the overall structure and delegation for Ingress occurring now?  If things are stabilized on that front (love the model) I can take a look at the code base.

**Comment by davecheney** [comment]
Things should be stable for a while.

**Comment by davecheney** [comment]
Duplicate of #403. Will follow up on #403
